### PR TITLE
Test changes

### PR DESCRIPTION
**Pull request by mbolivar-nordic**
Miscellaneous commits related to testing.

### CODE REVIEW

**Review by marc-hb — COMMENTED** [review]
While `split up testenvs` does solve a real test selection issue that I actually experienced myself, I must say it looks like a hack to me. `tox` is "a generic virtualenv management", not a test runner and this is very much subverting test environments to act as test selectors.  This being said, it does the job (fetched and tested!) and I have no better solution to offer. `pytest` is the test runner with good selection abilities but I guess we don't want to run flake8 and mypy from `pytest`?

Can you maybe add one or two comments in the `tox.ini` file admitting this is not what `tox` is normally designed for?

One immediate and relatively harmless consequence is that running a basic, plain `tox` as before will run zero test and print a green "success" message with even a smiley on top. As noted offline by @mbolivar-nordic, CI will catch any issue anyway and people should learn soon enough to stop doing that.

**Comment by mbolivar-nordic** [comment]
> `tox` is "a generic virtualenv management", not a test runner

I don't agree with that. Tox's [front page](https://tox.readthedocs.io/en/latest/index.html) says:

> What is tox?
>
> tox is a generic virtualenv management and test command line tool

**Comment by marc-hb** [comment]
I had read the rest of the page too :-) It's all about test environments and I found nothing about test selection there. There's that too:

> 3. environment - for each tox environment (e.g. py27, py36) do:
> ...
> run the specified commands in the specified order.

Almost every example on the entire tox site runs the same tests in each environment. Not a problem when the `commands` is actually just invoking `pytest`; then test selection can just be deferred to `pytest`. Much more annoying when listing totally different sorts of test in the same `tox.ini` file. Speaking of which: have you considered multiple `tox.ini` files? The current tests seem to share barely anything between them in the current, unique `tox.ini` file.

**Comment by mbolivar-nordic** [comment]
> I had read the rest of the page too :-) It's all about test environments and I found nothing about test selection there.

That's a fair point. I've dropped that patch for now. I think the other two are OK though.

File: tests/test_config.py [hunk 1]
```diff
@@ -13,6 +13,8 @@
 
 from conftest import cmd
 
+assert 'TOXTEMPDIR' in os.environ, "you must run these tests using tox"
```
[the remark this answers is not in the document]
Comment (mbolivar-nordic) [reply tests/test_config.py:16]: > Can you add a comment alluding to why?

I've added a comment in tox.ini in a separate commit to avoid duplication.

**Review by marc-hb — APPROVED** [review]
Note there is still mention of `tox -e ALL` in one commit message.

**Comment by mbolivar-nordic** [comment]
> Note there is still mention of `tox -e ALL` in one commit message.

Thanks for catching that.

**Comment by mbolivar-nordic** [comment]
I added one more: `ci: test on 3.9 too`

**Review by carlescufi — APPROVED** [review]

**Pull request merged**